components/provisioner/internal/operations/stages/provisioning/wait_for_agent_to_connect.go

```diff
@@ -92,7 +98,12 @@ func (s *WaitForAgentToConnectStep) Run(cluster model.Cluster, _ model.Operation
 	}
 
 	if compassConnCR.Status.State == v1alpha1.ConnectionFailed {
-		return operations.StageResult{}, fmt.Errorf("error: Compass Connection is in Failed state")
+		log.Warn("Compass Connection is in Failed state, trying to reconfigure runtime")
+		err := s.runtimeConfigurator.ConfigureRuntime(cluster, *cluster.Kubeconfig)
+		if err != nil {
+			return operations.StageResult{}, fmt.Errorf("error: Compass Connection is in Failed state: reconfigure runtime faiure: %s", err.Error())
+		}
+		return operations.StageResult{Stage: s.Name(), Delay: 2 * time.Minute}, nil
```

Comment:
This is OK, however I would personally make this delay a little bigger like 5 minutes...